thentos-core/purescript/src/Main.purs

```diff
@@ -0,0 +1,68 @@
+module Main where
+
+import Control.Monad.Aff
+import Control.Monad.Eff
+import Control.Monad.Eff.Class
+import Control.Monad.Eff.Console (CONSOLE(), log, print)
+import Control.Monad.Eff.Exception
+import Data.Array (zipWith, range, length)
+import Data.Either
+import Data.Foreign.Class
+import Data.Foreign hiding (parseJSON)
+import Data.Generic
+import Data.JSON
+import Data.Maybe
+import Data.Tuple
+import Network.HTTP.Affjax
+import Network.HTTP.Affjax.Request
+import Network.HTTP.Affjax.Response
+import Network.HTTP.Method
+import Network.HTTP.MimeType
+import Network.HTTP.MimeType.Common
+import Network.HTTP.RequestHeader
+import Network.HTTP.StatusCode (StatusCode(StatusCode))
+import Prelude
+
+
+defRq :: AffjaxRequest Unit
+defRq = defaultRequest { headers = [ContentType applicationJSON, Accept applicationJSON] }
+
+type Username = String
+type Password = String
+type ThentosSessionToken = String
+
+
+data ThentosError = ConnectionError String | DecodeError String
+
+derive instance genericThentosError :: Generic ThentosError
+instance showThentosError :: Show ThentosError where
+    show = gShow
+
+data LoginRequestBody = LoginRequestBody
+    { user :: Username
+    , pass :: Password
+    }
+
+instance loginRequestBodyToJSON :: ToJSON LoginRequestBody where
+  toJSON (LoginRequestBody { user: n, pass: p }) = object ["ldName" .= n, "ldPassword" .= p]
```

Comment:
@fisx why `ld`? Seems like an artefact of the backend not wanting record name conflicts?